script.js

```diff
@@ -1,129 +1,159 @@
-    $(this).attr("data-disabled", true).closest("form").submit();
+  var requestMarkAsSolvedButton = document.querySelector('.request-container .mark-as-solved:not([data-disabled])'),
+    requestMarkAsSolvedCheckbox = document.querySelector('.request-container .comment-container input[type=checkbox]'),
+    requestCommentSubmitButton = document.querySelector('.request-container .comment-container input[type=submit]');
+
+  requestMarkAsSolvedButton && requestMarkAsSolvedButton.addEventListener('click', function () {
+    requestMarkAsSolvedCheckbox.setAttribute('checked', true);
+    requestCommentSubmitButton.disabled = true;
+    this.setAttribute('data-disabled', true);
+    // Element.closest is supported in modern browsers and has a polyfill available for still in use outdated versions
+    // See https://caniuse.com/#feat=element-closest
+    // TODO: Should polyfill be installed? 
+    closest(this, 'form').submit();
   });
 
   // Change Mark as solved text according to whether comment is filled
-  var $requestCommentTextarea = $(".request-container .comment-container textarea");
+  var requestCommentTextarea = document.querySelector('.request-container .comment-container textarea');
 
-  $requestCommentTextarea.on("input", function() {
-    if ($requestCommentTextarea.val() !== "") {
-      $requestMarkAsSolvedButton.text($requestMarkAsSolvedButton.data("solve-and-submit-translation"));
-      $requestCommentSubmitButton.prop("disabled", false);
+  requestCommentTextarea && requestCommentTextarea.addEventListener('input', function() {
+    if (requestCommentTextarea.value !== '') {
+      requestMarkAsSolvedButton.innerText = requestMarkAsSolvedButton.getAttribute('data-solve-and-submit-translation');
+      requestCommentSubmitButton.disabled = false;
     } else {
-      $requestMarkAsSolvedButton.text($requestMarkAsSolvedButton.data("solve-translation"));
-      $requestCommentSubmitButton.prop("disabled", true);
+      requestMarkAsSolvedButton.innerText = requestMarkAsSolvedButton.getAttribute('data-solve-translation');
+      requestCommentSubmitButton.disabled = true;
     }
   });
 
   // Disable submit button if textarea is empty
-  if ($requestCommentTextarea.val() === "") {
-    $requestCommentSubmitButton.prop("disabled", true);
+  if (requestCommentTextarea && requestCommentTextarea.value === '') {
+    requestCommentSubmitButton.disabled = true;
   }
 
   // Submit requests filter form in the request list page
-  $("#request-status-select, #request-organization-select")
-    .on("change", function() {
+  [].forEach.call(document.querySelectorAll('#request-status-select, #request-organization-select'), function(el) {
+    el.addEventListener('change', function() {
       search();
     });
+  });
 
   // Submit requests filter form in the request list page
-  $("#quick-search").on("keypress", function(e) {
-    if (e.which === 13) {
+  const quickSearch = document.querySelector('#quick-search')
```

Comment:
Missing semicolon for consistency.

Reply:
still missing?

Reply:
Also, we can use `const` or any `es6` features here we're not transpiling this this file.

Reply:
`const` is supported in the supported browsers but simple treated as `var`. Do you suggest I replace all occurrences with `var`?